lib/_stream_readable.js

```diff
@@ -1070,13 +1073,74 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
-  if (createReadableStreamAsyncIterator === undefined) {
-    createReadableStreamAsyncIterator =
-      require('internal/streams/async_iterator');
+  let stream = this;
+
+  if (typeof stream.read !== 'function') {
+    // v1 stream
+
+    if (!eos) eos = require('internal/streams/end-of-stream');
+
+    const src = stream;
+    stream = new Readable({ objectMode: true }).wrap(src);
+    eos(stream, (err) => destroyImpl.destroyer(src, err));
   }
-  return createReadableStreamAsyncIterator(this);
+
+  const iter = createAsyncIterator(stream);
+  iter.stream = stream;
+  return iter;
 };
 
+async function* createAsyncIterator(stream) {
+  const state = stream._readableState;
+
+  if ((state && state.endEmitted) ||
+      (state && state.closeEmitted)) {
+    stream.on('error', nop);
+    return;
+  }
+
+  let _callback = nop;
+  let _error;
```

Comment:
Could you drop the leading underscore on these?

Reply:
done